### PR TITLE
fixing report output, cleaning up logging, use mochawesome assetsDir

### PR DESCRIPTION
**Pull request by grawk**
- fix logic around `output.reports`
  - avoid writing `summary.json` to undefined directory
  - fallback to manually configured mochawesome/xunit if `output.reports` DNE
  - don't create timestamped directories if `output.reports` DNE
  - don't take screenshots if `snap()` has no path specified and `output.reports` DNE
- clean up some error/info logging for consistency
- integrate new [mochawesome-report-generator assetsDir feature](https://github.com/adamgruber/mochawesome-report-generator/blob/master/CHANGELOG.md#added-1)
- rename modules to be more akin to their purpose
- add first revision of graphQL endpoint

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -57,11 +57,15 @@ $ ./bin/nemo --help
 
 ### `output`
 
-### `output.reports`
+### `output.reports <optional>`
+
+This convenience setting will create timestamped and tag-based directory structure for reports and
+screenshots when you use [mochawesome](https://www.npmjs.com/package/mochawesome) or [xunit](https://mochajs.org/#xunit) reporters. When you use this, you can omit the specific
+directory/filename settings for those reporters, as nemo will take care of that for you.
```
Comment (ethangodt) [inline README.md]: What are the benefits of `mochawesome` and `xunit` respectively? I'm hoping that the `nemo-reporter` I'm integrating can be a default (but not a mandate) and can save teams from setting up 2+ reporters for different environments. So far, it's worked better for the people that I've onboarded around consumer than those tools both locally and on CI.

I don't know a lot about those tools, though. If there are good reasons to keep encouraging them, ideally I would just add those good reasons/features to nemo-reporter and the docs can be made more simple (where a new user doesn't have to worry about setting up a reporter at all).

File: README.md [hunk 2]
```diff
@@ -57,11 +57,15 @@ $ ./bin/nemo --help
...
 
 Recommended to set this as `path:report`, which will create a `report` directory beneath your base directory. See `Reporting` below.
 
-### `output.storage <optional but cool>`
+### `output.storage <optional>`
```
Comment (ethangodt) [inline README.md]: still very cool :)

File: bin/_nemo [hunk 3]
```diff
@@ -15,7 +15,7 @@ var log;
 var error;
 var server;
 var scaffold;
-var prefontaine;
+var engine;
```
Comment (ethangodt) [inline bin/_nemo]: 👍 love it. In hindsight I can see now how the runner analogy can work, but more boring names like this seem better for grok-ing the code quickly and making contributions.

File: bin/_nemo [hunk 4]
```diff
@@ -65,9 +65,9 @@ if (program.scaffold) {
   return scaffold(program);
 }
 log('CLI launching runner');
-prefontaine(program)
-  .then(function (pre) {
-    return prefontaine.start(pre);
+engine.configure(program)
+  .then(function (configuration) {
```
Comment (ethangodt) [inline bin/_nemo]: 👍 feels much better

File: README.md (hunk 1 above)
Comment (grawk) [reply README.md]: I agree. Do you have an ETA on being able to get the reporter out in open source?

Comment (ethangodt) [reply README.md]: I'll have a PR tonight, I want to try to marry it from this repo. I want to see how that feels. We can go from there if it feels right.

**Pull request merged**